// // Exercice 7: Manipuler un Result d'une promesse et rÃ©cupÃ©rer la valeur avec une exceptionFonctions Ã  utiliser : fromPromise, map, getExnObjectif : Transformer une promesse en Result, appliquer une transformation, puis rÃ©cupÃ©rer la valeur avec exception si le Result Ã©choue.

import { R } from "@mobily/ts-belt";

const fetchData = async () => {
  if (Math.random() > 0.5) {
    return "DonnÃ©es";
  }
  throw new Error("Erreur");
};

export const resultFn7 = async () => {
  const result = await R.fromPromise(fetchData());
  const transformedResult = R.map(result, (data: string) => data.toUpperCase());

  try {
    const value = R.getExn(transformedResult);
    console.log("ðŸš€ ~ resultFn7 ~ value:", value)
  } catch (err) {
    console.log("ðŸš€ ~ resultFn7 ~ err:", err)
  }
};

resultFn7();
